packages/glimmer-test-helpers/lib/environment.ts

```diff
@@ -80,14 +80,14 @@ import {
   UpdatableReference
 } from "glimmer-object-reference";
 
-type KeyFor = (item: Opaque, index: number) => string;
+type KeyFor = (item: Opaque, index: number | string) => string;
```

Comment:
Since this is only used by the tests, I'm not too sad about it. We can probably make it `type keyFor<T> = (item: Opaque, memo: T) => string;`. Then `ArrayIterator` can have a `KeyFor<number>`, and `ObjectKeysIterator` can have a `KeyFor<string>`.